CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+# Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.13.4)
+
+project(Chapel VERSION 1.29.0)
```

Comment:
We can store this version in config.h.cmake & replace some of the version logic with it.

A TODO is how to get the git sha into it. Maybe we could save the git sha at the time we `cmake` it?

Reply:
I think the approach outlined here would be our best bet, just based on what they are saying about recompiling and when the SHA gets updated etc. (Not necessarily make this a dependency, just add some of the functionality to our project)
https://github.com/andrew-hardin/cmake-git-version-tracking/

Reply:
I have some ideas about how to handle the git SHA.

First, I actually think it's really annoying to re-link when we change the SHA. E.g. if you are working on module code, documentation, or the runtime and doing commits, relinking the compiler just takes time & doesn't matter. Second, we already have some logic in version.cpp (see `get_version`) that does different things in release mode.

So here is what I am imagining:
 1. We move this version.cpp logic to dyno (as a util/ thing). It will give the "version of the dyno library" which should also match the version of Chapel everything since it's all one repo. That way, we can use it from both `chpl` and `chpldoc`.
 2. For release builds, the git sha is not included anyway, so there is no point in trying to link it in to the binary. However, release builds do need the ability to manually update the build version, which happens today by updating the BUILD_VERSION file. But, today the BUILD_VERSION file sometimes contains a number like 0 1 2 etc and sometimes contains a git sha. In other work we were discussing that it might be a good idea to separate out the git sha from BUILD_VERSION. So the proposal here is that BUILD_VERSION continue to exist but only be used for release builds; and the BUILD_VERSION file never contains a git sha.
 3. That leaves the question of what to do for non-release builds. Here what I am imagining to be the desired behavior is:
   * When the compiler is re-built, the git sha printed by the version matches the git sha of the repo used when building it.
   * When such a compiler build is installed somewhere, this git sha should still be in the version output

So, to achieve the behavior in 3, without requiring any re-linking due to changes to the git sha only, I am proposing this:
 * In a directory that we install & that contains build products (e.g. `lib`), we put a file like `gitversion` that just contains the git sha. If it helps, we could put this file next to `liblibdyno.so`.
 * When doing `--version` for a non-release build, we find this file and output its contents. We can just ignore it if the file is missing.

Another possibility would be to only use such a file for the case that `chpl` has not been installed and use a `cmake` header value when it is installed. But, I think the little file approach can work OK for us.